vulnfeeds/cves/versions.go

```diff
@@ -88,50 +97,82 @@ func (vi *VersionInfo) HasLastAffectedVersions() bool {
 }
 
 func (vi *VersionInfo) HasIntroducedCommits(repo string) bool {
-	for _, av := range vi.AffectedCommits {
-		if av.Repo == repo && av.Introduced != "" {
+	for _, ac := range vi.AffectedCommits {
+		if ac.Repo == repo && ac.Introduced != "" {
 			return true
 		}
 	}
 	return false
 }
 
 func (vi *VersionInfo) HasFixedCommits(repo string) bool {
-	for _, av := range vi.AffectedCommits {
-		if av.Repo == repo && av.Fixed != "" {
+	for _, ac := range vi.AffectedCommits {
+		if ac.Repo == repo && ac.Fixed != "" {
 			return true
 		}
 	}
 	return false
 }
 
 func (vi *VersionInfo) HasLastAffectedCommits(repo string) bool {
-	for _, av := range vi.AffectedCommits {
-		if av.Repo == repo && av.LastAffected != "" {
+	for _, ac := range vi.AffectedCommits {
+		if ac.Repo == repo && ac.LastAffected != "" {
 			return true
 		}
 	}
 	return false
 }
 
 func (vi *VersionInfo) FixedCommits(repo string) (FixedCommits []string) {
-	for _, av := range vi.AffectedCommits {
-		if av.Repo == repo && av.Fixed != "" {
-			FixedCommits = append(FixedCommits, av.Fixed)
+	for _, ac := range vi.AffectedCommits {
+		if ac.Repo == repo && ac.Fixed != "" {
+			FixedCommits = append(FixedCommits, ac.Fixed)
 		}
 	}
 	return FixedCommits
 }
 
 func (vi *VersionInfo) LastAffectedCommits(repo string) (LastAffectedCommits []string) {
-	for _, av := range vi.AffectedCommits {
-		if av.Repo == repo && av.LastAffected != "" {
-			LastAffectedCommits = append(LastAffectedCommits, av.Fixed)
+	for _, ac := range vi.AffectedCommits {
+		if ac.Repo == repo && ac.LastAffected != "" {
+			LastAffectedCommits = append(LastAffectedCommits, ac.Fixed)
 		}
 	}
 	return LastAffectedCommits
 }
 
+// Check if the same commit appears in multiple elements of the AffectedCommits array.
+// See https://github.com/google/osv.dev/issues/1984 for more context.
+func (vi *VersionInfo) Duplicated(candidate AffectedCommit) (err error) {
```

Comment:
nit:  is the `err` expected to be used anywhere? it doesn't appear to be used above. If it's not used, I suggest following YAGNI and just returning a bool.

Reply:
Addressed in [0bfb1a4](https://github.com/google/osv.dev/pull/2127/commits/0bfb1a4d4902cf9fbf7593f03863dd37b8724ccb)